.github/workflows/python-tests.yml

```diff
@@ -76,7 +76,26 @@ jobs:
       - name: Run Tests with Coverage
         run: |
           pipenv run coverage run -m pytest --cov=./src --cov-report=xml
+      - uses: actions/upload-artifact@v4
+        with:
+          name: coverage-report-${{ matrix.os }}
+          path: ./coverage.xml
+
+  python-tests-codecov:
+    needs: python-tests-linux
+    strategy:
+      matrix:
+        os:
+          - ubuntu-24.04-arm  # arm64-preview
+          - ubuntu-24.04  # x64
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/download-artifact@v4
```

Comment:
Should be up `upload-artifact`?

Reply:
No, because we are downloading the coverage.xml that we uploaded from the actual pytest run job section.

Reply:
Also we dont need to upload twice. It will just overwrite the previous results right?

Reply:
Sorry not getting why we are downloading the coverage?

Reply:
Because AL2 does not support codecov actions over v3.
And only v5 have fixed this issue.

So we need to run the test on AL2, upload the coverage to github artifact, download it in a normal ubuntu server, then send to codecov using v5.

Reply:
Ah okay! In that case, why run the matrix if we are just uploading the already generated report?

Reply:
Because arm64 and x64 have two coverage yml, not one.
So I just for the sake of it upload twice, through matrix, instead of manually write two similar steps.

Reply:
I dont think it makes sense to upload the coverage twice. The tests are same and when you upload the coverage to the codecov it is for per branch. So the results will be overwritten whoever wins the race. Please correct if I am wrong.

Reply:
It is just to avoid hardcoding, upload twice is very common even in the original setup.
It wont cause any issue and codecov can easily get around it.

Reply:
Better to save resources for actual tests than using for minor uploads? Does not give an value IMO.

Reply:
It is possible at times where one test on x64 success while arm64 fail, so I dont think this is necessary.

Reply:
In that case code coverage will not be uploaded as the step abruptly exits? Only on successful runs, code coverage gets uploaded to the codecov. [Example](https://github.com/opensearch-project/opensearch-build/pull/5263#issuecomment-2613640365)

Reply:
Updated to use only one upload.